bigquery/monitor-tables-for-errors.sql

```diff
@@ -5,52 +5,70 @@ SELECT
+            (table LIKE "%vacancy"
+              AND row_count <29000,
+              #sets a minimum threshold for the number of schools that should be in the database - this should not change much unless we change the scope, so going below this threshold most likely means the dataset in BQ is incomplete
               "Table appears incomplete",
-              #check whether there are at least this many approvers in the dsi_approvers table
+              #check whether there are at least this many schools in the schools table
             IF
-              ((table ="dsi_users"
-                  AND (
-                  SELECT
-                    MAX(update_datetime)
-                  FROM
-                    `teacher-vacancy-service.production_dataset.dsi_users` ) < TIMESTAMP_SUB(CURRENT_TIMESTAMP(),INTERVAL 3 DAY)) #check the last updated date for any user's data from DSI - produce error if more than three days old (i.e. if longer ago than the last working day)
-                OR (table LIKE "feb20_%"
-                  AND (
-                  SELECT
-                    MAX(run_on)
-                  FROM
-                    `teacher-vacancy-service.production_dataset.feb20_alertrun` ) < DATE_SUB(CURRENT_DATE(),INTERVAL 3 DAY)),
-                "Latest date a row was updated in a frequently updated table from this source was at least 3 days ago",
-                "OK"))))))) AS status
+              (table = "dsi_users"
+                AND row_count <25000,
+                #sets a minimum threshold for the number of users that should be in the DSI database - this should not decrease, so going below this threshold most likely means the dataset in BQ is incomplete
+                "Table appears incomplete",
+                #check whether there are at least this many users in the dsi_users table
+              IF
+                (table = "dsi_approvers"
+                  AND row_count <35000,
+                  #sets a minimum threshold for the number of approvers that should be in the DSI database - this should not decrease, so going below this threshold most likely means the dataset in BQ is incomplete
+                  "Table appears incomplete",
+                  #check whether there are at least this many approvers in the dsi_approvers table
+                IF
+                  ((table ="dsi_users"
+                      AND (
+                      SELECT
+                        MAX(update_datetime)
+                      FROM
+                        `teacher-vacancy-service.production_dataset.dsi_users` ) < TIMESTAMP_SUB(CURRENT_TIMESTAMP(),INTERVAL 3 DAY)) #check the last updated date for any user's data from DSI - produce error if more than three days old (i.e. if longer ago than the last working day)
+                    OR (table LIKE "feb20_%"
```

Comment:
feb20_% is this correct? seems a bit hardcoded

Reply:
this isnt strictly speaking a change you have made so approving this, but looks strange

Reply:
Thanks - it's because we don't have another way to identify which tables came from the database in BigQuery right now.